### PR TITLE
Eliminate more "multiple expectations" warnings and remove some redundant specs

### PR DESCRIPTION
**Pull request by wyardley**
Would appreciate feedback on this one.

### CODE REVIEW

File: spec/unit/puppet/type/rabbitmq_policy_spec.rb [hunk 1]
```diff
@@ -171,13 +167,25 @@
...
   end
 
-  it 'accepts list value in ha-params when ha-mode = nodes' do
-    definition = { 'ha-mode' => 'nodes', 'ha-params' => ['rabbit@rabbit-01', 'rabbit@rabbit-02'] }
-    policy[:definition] = definition
-    expect(policy[:definition]['ha-mode']).to eq('nodes')
-    expect(policy[:definition]['ha-params']).to be_a(Array)
-    expect(policy[:definition]['ha-params'][0]).to eq('rabbit@rabbit-01')
-    expect(policy[:definition]['ha-params'][1]).to eq('rabbit@rabbit-02')
+  context 'accepts list value in ha-params when ha-mode = nodes' do
```
Comment (wyardley) [inline spec/unit/puppet/type/rabbitmq_policy_spec.rb]: I think this works Ok here, but not sure if there's a better way to do it.

File: spec/unit/puppet/type/rabbitmq_policy_spec.rb [hunk 2]
```diff
@@ -171,13 +167,25 @@
...
+    before do
+      policy[:definition] = definition
+    end
+
+    let(:definition) { { 'ha-mode' => 'nodes', 'ha-params' => ['rabbit@rabbit-01', 'rabbit@rabbit-02'] } }
+
+    it do
+      expect(policy[:definition]['ha-mode']).to eq('nodes')
+    end
```
Comment (alexjfisher) [inline spec/unit/puppet/type/rabbitmq_policy_spec.rb]: I might be tempted to use a single line for each of these?
ie.
```
it { expect(policy[:definition]['ha-mode']).to eq('nodes') }
```

Comment (wyardley) [reply spec/unit/puppet/type/rabbitmq_policy_spec.rb]: Will do. You think moving the outer bit to a context makes sense here though?

Comment (wyardley) [reply spec/unit/puppet/type/rabbitmq_policy_spec.rb]: Updated, btw, if you want to re-review

**Pull request merged**